File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        ns=len(needle)
        hs=len(haystack)
        for i in range(hs - ns + 1):
            if haystack[i:i + ns] == needle:
                return i
        return -1
        

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int ns=needle.size();
        int hs=haystack.size();
        int j=ns-1;
        for(int i=0;i<hs;i++)
        {
            if(haystack.substr(i,ns)==needle)return i;

        }
        return -1;
    }
};